lib/teleterm/daemon/daemon.go

```diff
@@ -212,6 +216,16 @@ func (s *Service) createGateway(ctx context.Context, params CreateGatewayParams)
 	return gateway, nil
 }
 
+func (s *Service) createCLICommandProvider(targetURI string) gateway.CLICommandProvider {
+	switch {
+	case uri.IsDB(targetURI):
+		return clusters.NewDbcmdCLICommandProvider(s.cfg.Storage, dbcmd.SystemExecer{})
```

Comment:
I think I made a mistake here, as far as specifying dependencies goes, the CLI command provider should have been initialized outside of `daemon.Service`, it should be received as a field on `Config` in `New`.

Which would mean that the code here would merely return an existing struct instead of making a new one.

Could you leave a todo comment here?

```
// TODO(ravicious): Construct command providers outside of daemon.Service and pass them as fields of Config in New.
```

Reply:
@ravicious @smallinsky here is my first refactor attempt to decouple dependency a bit:
https://github.com/gravitational/teleport/pull/27685